sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -846,6 +846,17 @@ class Column(val expr: Expression) extends Logging {
    */
   def rlike(literal: String): Column = withExpr { RLike(expr, lit(literal).expr) }
 
+  /**
+   * SQL ILIKE expression (case insensitive LIKE).
+   *
+   * @group expr_ops
+   * @since 3.3.0
+   */
+  def ilike(literal: String): Column = withExpr {
+    ILike(expr, lit(literal).expr, '\\',
+      Like(Lower(expr), Lower(lit(literal).expr), '\\'))
+  }
```

Comment:
```suggestion
  def ilike(literal: String): Column = withExpr { ILike(expr, lit(literal).expr) }
```

Reply:
Thank you, I changed here to use overloaded constructor.